joint_trajectory_controller/test/joint_trajectory_controller_test.cpp

```diff
@@ -1244,6 +1244,17 @@ TEST_F(JointTrajectoryControllerTest, pathToleranceViolation)
   EXPECT_TRUE(waitForState(action_client, SimpleClientGoalState::ABORTED, long_timeout));
   EXPECT_EQ(action_client->getResult()->error_code, control_msgs::FollowJointTrajectoryResult::PATH_TOLERANCE_VIOLATED);
 
+  // Check that we're not moving
+  StateConstPtr state1 = getState();
+  ros::Duration(0.5).sleep(); // Wait
+  StateConstPtr state2 =  getState();
+  for (unsigned int i = 0; i < n_joints; ++i)
+  {
+    EXPECT_NEAR(state1->desired.positions[i],     state2->desired.positions[i],     EPS);
```

Comment:
According to the failure log, the velocity in `state1` is non-zero, but in `state2` it is zero.
This shows that your path is working!

I would make the test check `state1` velocity and acceleration to be non-zero (=moving) and the ones in `state2` should be zero. In addition the position change between `state1` and `state2` should be in the range of the largest possible step in the hold-trajectory.

Reply:
There are a few other test cases that check for movement, for example [here](https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/joint_trajectory_controller/test/joint_trajectory_controller_test.cpp#L779). I made my check similar for consistency.

Reply:
These tests don't  have the delay enabled and a big delay before acquiring the states.

Reply:
Okay, I will update the test with your proposal.

Reply:
> I would make the test check state1 velocity and acceleration to be non-zero (=moving) and the ones in state2 should be zero. 

state1 velocity and acceleration are only non-zero, if `stop_trajectory_duration != 0`. I guess, I have to test for multiple cases here.

> In addition the position change between state1 and state2 should be in the range of the largest possible step in the hold-trajectory.

Do you know how to find this value? The actual implementation samples a trajectory.

Reply:
> I would make the test check state1 velocity and acceleration to be non-zero (=moving) and the ones in state2 should be zero. In addition the position change between state1 and state2 should be in the range of the largest possible step in the hold-trajectory.

Sounds good. I think we should strive for an assertion that reliably checks, if a hold trajectory is executed as expected. This could then be used in all relevant test cases. However, I would separate it from this PR.